File: Programing_Challenge.js
let arrayFromExercise = ["a", 10, "b", "hello", 122, 15];

function findNumbersLettersMax(arr) {
  let letters = [];
  let numbers = [];

  for (let i = 0; i < arr.length; i++) {
    let e = arr[i];
    if (typeof e === "string") {
      letters.push(e);
    } else if (typeof e === "number") {
      numbers.push(e);
    }
  }

  let largestInNumbers = Math.max(...numbers);

  return { letters, numbers, largestNumber: largestInNumbers };
}

let { letters, numbers, largestNumber } = findNumbersLettersMax(arrayFromExercise);

console.log("letters:", letters);
console.log("numbers:", numbers);
console.log("Largest number:", largestNumber);

File: Programing_Challenge.py
array_from_exercise = ["a", 10, "b", "hello", 122, 15]

def find_numbers_letters_max(arr):
    letters = []
    numbers = []
    
    for e in arr:
        if isinstance(e, str):
            letters.append(e)
        elif isinstance(e, int):
            numbers.append(e)
    
    largest_in_numbers = max(numbers)
    
    return letters, numbers, largest_in_numbers



letters, numbers, largest_number = find_numbers_letters_max(array_from_exercise)

print("letters:", letters)
print("numbers:", numbers)
print("Largest number:", largest_number)